provider/pkg/provider/diff.go

```diff
@@ -44,6 +49,138 @@ func getInputsFromState(res resources.CloudAPIResource, state resource.PropertyM
 	return resource.NewPropertyMapFromMap(inputsMap)
 }
 
+// calculateDetailedDiff produced a property diff for a given object diff and a resource definition. It inspects
```

Comment:
This is largely borrowed from azure-native. It would be ideal to leverage the replaceOnChanges support in the schema instead but a combination of https://github.com/pulumi/pulumi/issues/9158 and the fact that inputs are not a perfect subset of the properties causes this to be not ideal at the moment. The latter is likely something we need to fix anyway (codegen currently only looks at properties and not input properties) but the former would unblock broader usage of the schema's support.